app/controllers/assignments_controller.rb

```diff
@@ -92,7 +95,35 @@ def create
       @assignment.available = @assignment.due_date
     end
 
-    if @assignment.save
+    # Options are:
+    # None -- create a (unique, ignored) teamset
+    # New -- create a new teamset with initially no teams
+    # Use -- share an existing teamset with another assignment
+    # Copy -- duplicate an existing teamset
+    
+    @assignment.team_subs = (params["teamset"] != "none")
```

Comment:
Having 20 lines of code in the assignments controller about team set initialization can't be right.

I'd go with something like this:
 * Just use the team_set_id attribute to store the source ID, if any.
 * Have a team_set_init virtual attribute with the four possible values.
 * Have this logic in a before_save callback in the model.

This cleans up the controller, eliminates the duplicate code below, and since before_save supposedly happens in the DB transaction means the logic is transactional.